lib/matplotlib/__init__.py

```diff
@@ -1476,7 +1476,6 @@ def _jupyter_nbextension_paths():
 default_test_modules = [
     'matplotlib.tests.test_png',
     'matplotlib.tests.test_units',
-    'matplotlib.tests.test_widgets',
```

Comment:
I don't get it.

After the general conversion, do we expect `nosetests` to correctly run the tests?
- If the answer is yes, I give up (I appreciate your efforts in making this work but conversion to a test suite that supports two runners is way too hard for me to follow properly).
- If the answer is no, we already broke one way of running the tests.  I honestly cannot imagine why we would need to keep backcompat for another one, and would thus just kill `matplotlib.test()` (... and `tests.py` at the same time).